plugins/ui/twilight.nix

```diff
@@ -0,0 +1,68 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "twilight";
+    originalName = "twilight.nvim";
+    defaultPackage = pkgs.vimPlugins.twilight-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    settingsOptions = {
+      dimming = {
+        alpha = helpers.defaultNullOpts.mkNullable (types.numbers.between 0.0 1.0) "0.25" ''
+          Amount of dimming.
+        '';
+
+        color = helpers.defaultNullOpts.mkListOf types.str ''["Normal" "#ffffff"]'' ''
+          Highlight groups / colors to use.
+        '';
+
+        term_bg = helpers.defaultNullOpts.mkStr "#000000" ''
+          If `guibg=NONE`, this will be used to calculate text color.
+        '';
+
+        inactive = helpers.defaultNullOpts.mkBool false ''
+          When true, other windows will be fully dimmed (unless they contain the same buffer).
+        '';
+      };
+
+      context = helpers.defaultNullOpts.mkUnsignedInt 10 ''
+        Amount of lines we will try to show around the current line.
+      '';
+
+      treesitter = helpers.defaultNullOpts.mkBool true ''
```

Comment:
Shouldn't we warn if this option is enabled but treesitter is not?

Reply:
Indeed, I added a warning.